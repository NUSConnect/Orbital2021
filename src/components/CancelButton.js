import React from 'react'
import { StyleSheet, Text, TouchableOpacity } from 'react-native'

export default function CancelButton ({ goBack }) {
  return (
    <TouchableOpacity onPress={goBack} style={styles.container}>
      <Text style={styles.text}>Cancel</Text>
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#dcdcdc',
    alignItems: 'center',
    height: 50,
    width: 180,
    borderRadius: 30
  },
  text: {
    color: '#ffffff',
    fontWeight: 'bold',
    fontSize: 30,
    lineHeight: 50,
    alignItems: 'center'
  }
})
