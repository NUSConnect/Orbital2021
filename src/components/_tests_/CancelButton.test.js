import * as React from 'react'
import CancelButton from '../CancelButton'
import renderer from 'react-test-renderer'
import { fireEvent, render } from '@testing-library/react-native'

jest.useFakeTimers()

it('renders correctly', () => {
  const tree = renderer.create(<CancelButton />)
  expect(tree).toMatchSnapshot()
})

it('on press button', () => {
  const mockCancelFn = jest.fn()
  const { getByTestId } = render(<CancelButton goBack={mockCancelFn} />)
  fireEvent.press(getByTestId('cancel'))
  expect(mockCancelFn).toHaveBeenCalled()
})
