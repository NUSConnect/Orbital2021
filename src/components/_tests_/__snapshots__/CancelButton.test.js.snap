// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#dcdcdc",
      "borderRadius": 30,
      "height": 50,
      "opacity": 1,
      "width": 180,
    }
  }
>
  <Text
    style={
      Object {
        "alignItems": "center",
        "color": "#ffffff",
        "fontSize": 30,
        "fontWeight": "bold",
        "lineHeight": 50,
      }
    }
  >
    Cancel
  </Text>
</View>
`;
